latch_cli/click_utils.py

```diff
@@ -51,7 +143,84 @@ def format_epilog(self, ctx: Context, formatter: HelpFormatter) -> None:
         )
 
 
-class LatchGroup(LatchCommand, Group): ...
+
+----
+class EnumChoice(Choice):
+    def __init__(self, choices: Type[Enum], case_sensitive: bool = True):
+        self.enum = choices
+        return super().__init__(choices._member_names_, case_sensitive)
+
+    def convert(
+        self, value: Any, param: Optional[Parameter], ctx: Optional[Context]
+    ) -> Any:
+        ret = super().convert(value, param, ctx)
+        return self.enum(ret)
+
+    # todo(ayush): override `shell_complete` once we support it
+
+
+class ColoredHelpFormatter(HelpFormatter):
+    def write_usage(
+        self, prog: str, args: str = "", prefix: Optional[str] = None
+    ) -> None:
+        if prefix is None:
+            prefix = "Usage: "
+
+        prefix = style(prefix, fg="bright_blue")
+        return super().write_usage(prog, args, prefix)
+
+    def write_heading(self, heading: str):
+        return super().write_heading(style(heading, fg="bright_blue"))
+
+    def write_dl(
+        self, rows: Sequence[Tuple[str, str]], col_max: int = 30, col_spacing: int = 2
+    ) -> None:
+        return super().write_dl(
+            [(style(x[0], fg="bright_green", italic=True), x[1]) for x in rows],
+            col_max,
+            col_spacing,
+        )
+
+
+class LatchCommand(Command):
+    def format_epilog(self, ctx: Context, formatter: HelpFormatter) -> None:
+        formatter.write_paragraph()
+        formatter.write_text(
+            style("See " + style("https://docs.latch.bio/", underline=True), dim=True)
+            + style(" for manuals, tutorials, examples, and an API reference", dim=True)
+        )
+
+
+class LatchGroup(LatchCommand, Group):
+    def get_command(self, ctx, cmd_name):
+        rv = Group.get_command(self, ctx, cmd_name)
+        if rv is not None:
+            return rv
+        
+        # check for exact matches
+        matches = [
+            x for x in self.list_commands(ctx) 
+            if x == cmd_name
+        ]
+        if len(matches) > 1:
+            ctx.fail(f"Too many matches: {', '.join(sorted(matches))}")
+        elif len(matches) == 1:
+            return Group.get_command(self, ctx, matches[0])
+        
+        # check for similarity the same way GIT does
+        matches = _find_similar(cmd_name, self.list_commands(ctx))
+        if len(matches) == 0:
+            return None
+        word = "this" if len(matches) == 1 else "one of these"
+        option_sep = "\n        "
+        ctx.fail(
+            f"No such command '{cmd_name}'\nDid you mean {word}?{option_sep}" + option_sep.join(matches)
+        )
+
+    def resolve_command(self, ctx, args):
+        # always return the full command name
+        _, cmd, args = super().resolve_command(ctx, args)
+        return cmd.name, cmd, args
```

Comment:
can you guarantee that overriding this doesn't have any undesired side-effects? in testing locally removing this override didn't change anything, so i doubt this is necessary in the first place